tests/extensions/test_custom.py

```diff
@@ -145,7 +134,74 @@ def setUp(self) -> None:
     def tearDown(self) -> None:
         pystac.EXTENSION_HOOKS.remove_extension_hooks(SCHEMA_URI)
 
-    # TODO: Test custom extensions and extension hooks
+    def test_add_to_item_asset(self) -> None:
+        item = Item("an-id", None, None, datetime.datetime.now(), {})
+        item.add_asset("foo", Asset("http://pystac.test/asset.tif"))
+        custom = CustomExtension.ext(item.assets["foo"], add_if_missing=True)
+        CustomExtension.has_extension(item)
```

Comment:
Assuming you meant to prefix this with an `assert`? Otherwise, this line does nothing. Same comment for following tests.